### ISSUE TITLE
What happened to --no-exit option?

### ISSUE THREAD

**Issue by dougwilson** [mochajs/mocha#3475]
### Prerequisites

- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [x] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend avoiding the use of globally installed Mocha.

### Description

It seems that the `--no-exit` command line argument went missing at some point. I see it is still documented at https://mochajs.org/#--exit----no-exit but doesn't exist in the current version of mocha: 5.2.0.

### Steps to Reproduce

```
$ npx mocha@5.2.0 --no-exit
npx: installed 24 in 4.714s

  error: unknown option `--no-exit'
```

**Expected behavior:** Mocha runs tests with the no exit (default) behavior.

**Actual behavior:** Error that there is no `--no-exit` option.

**Reproduces how often:** 100%

### Versions

- The output of `mocha --version` and `node node_modules/.bin/mocha --version`: 5.2.0
- The output of `node --version`: v8.12.0
- The version and architecture of your operating system: Win 10 64 bit
- Your shell (bash, zsh, PowerShell, cmd, etc.): bash
- Your browser and version (if running browser tests): N/A
- Any other third party Mocha related modules (with versions): N/A
- The code transpiler being used: N/A

**Comment by plroebuck** [comment]
As of Mocha-5.2.0, the default was switched from '--exit' to '--no-exit' in [this PR](https://github.com/mochajs/mocha/commit/3fdd3ff165170c5b3c4da4edc3edc8270261c62d).
Unsure why documentation was not updated to match cmdline arguments, but the option is now implicit unless `--exit` is provided.

@ScottFreeCode, have anything to add here?

**Comment by plroebuck** [comment]
Two options to correct issue:
* Add `--no-exit` cmdline argument back to "_mocha" [here](https://github.com/mochajs/mocha/blob/d401ee198bd5026bc61641ad26eab31e6746713f/bin/_mocha#L206), but leave it unused (noop).
    -or-
* Need to remove references to `--no-exit` (and maybe reword a sentence or so) [here](https://github.com/mochajs/mocha/blob/master/docs/index.md).

**Comment by outsideris** [comment]
I believe the change is [this commit](https://github.com/mochajs/mocha/commit/3fdd3ff165170c5b3c4da4edc3edc8270261c62d). `--no-exit` option is replaced with `--exit` because the default behavior is changed. The document is a little bit confused. (I voted to second options which @plroebuck suggest)

BTW if you want to v5.x behave like v2.x, use `--exit`. It is default behavior on prior to v4.x.

**Comment by dougwilson** [comment]
> BTW if you want to v5.x behave like v2.x, use --exit. It is default behavior on prior to v4.x.

I want the opposite: 3.x to behave like 5.x . The system I have could have either installed and I would have liked to just call `mocha --no-exit` and both 3.x and 5.x work the same. Instead when I do this 3.x behaves like 5.x and 5.x throws an error.

I have to resort to version sniffing mocha now.

**Comment by plroebuck** [comment]
@dougwilson, any problems with patch?

**Comment by dougwilson** [comment]
HI @plroebuck I haven't tested it yet. I don't have direct access to the systems and so will need to put together a script that will patch the existing mocha installs.

**Comment by plroebuck** [comment]
@dougwilson, any problems with patch?

**Comment by boneskull** [comment]
This seems like a documentation issue.

**Comment by boneskull** [comment]
...actually... I figured commander would allow `--no-` as the prefix for any boolean option so am kind of surprised this is broken.  This is a bug if you think a my ignorance or violation of my expectations is a bug.  :smile:

**Comment by boneskull** [comment]
yeah, from the commander docs:

> Note that multi-word options starting with --no prefix negate the boolean value of the following word. For example, --no-sauce sets the value of program.sauce to false.

So wtf is going on...

**Comment by boneskull** [comment]
okay, I get it, it *doesn't add support for both* `--foo` and `--no-foo`, it just sets some value to `true` or `false`, respectively.

**Comment by boneskull** [comment]
OK, so this issue is "fix the docs".  I'll create another issue for dealing with those flags, which may end up being a dupe of #3037.

**Comment by boneskull** [comment]
@dougwilson I'll see about getting this fixed, but will likely be in v6.0.0 (should be next release)

**Comment by dougwilson** [comment]
Hi, sorry I forgot to follow up here. The patch did work. I ended up just removing the usage of the flag from my tests. The context is that many employees will run a test suite. The suite runs on various different versions of mocha for reasons I don't have control over. I wanted to get them to have the no exit behavior across the board, but if I include the flag most people cannot run them (as most use the current version of mocha) since it throws an error.

The only time it really comes into play is when a human edits the tests, as that is when a non exit behavior may be introduced. For old mochas those devs won't catch it until CI since the flag is not there.

I figured it was strange the docs seemed like it should work so opened an issue.

I hope that makes sense and look forward to the release you mentioned :)
